### PR TITLE
Add power function for symmetric positive definite matrices

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

File: mat64/symmetric_test.go [hunk 1]
```diff
@@ -623,3 +623,48 @@ func TestViewGrowSquare(t *testing.T) {
...
+				-0.612153996497505, 1.432842761381494, 4.695873060862573,
+			}),
+		},
+	} {
+		var s SymDense
+		err := s.PowPSD(test.a, test.pow)
+		if err != nil {
+			panic("bad test")
+		}
+		if !EqualApprox(&s, test.ans, 1e-10) {
```
Comment (kortschak) [inline mat64/symmetric_test.go]: Is it worth adding integer comparison cases against `*Dense.Pow`?

**Comment by btracey** [comment]
General non-integer Pow cases may have complex values. 

We would have
````
Dense.Pow(a, int)
SymDense.PowSym(s,int)
SymDense.PowPSD(s, float64)
CDense.Pow(a, float64)
CSymDense.PowSym(s,float64)
````

If that sounds good then I'll work on the rest.

**Comment by kortschak** [comment]
SGTM

**Comment by btracey** [comment]
I'll fix the things you mention, but I also found a bug in equalApprox where everything compared to true.

File: mat64/symmetric.go [hunk 2]
```diff
@@ -466,3 +469,32 @@ func (s *SymDense) GrowSquare(n int) Matrix {
 	v.cap = s.cap
 	return &v
 }
+
+// PowPSD computes a^pow where a is a positive symmetric definite matrix.
+//
+// PowPSD returns an error if the matrix is not PSD.
```
Comment (kortschak) [inline mat64/symmetric.go]: "... if the matrix is not positive symmetric definite or no Eigendecomposition is possible."?

Comment (btracey) [inline mat64/symmetric.go]: Did a slight variant.

File: mat64/symmetric.go [hunk 3]
```diff
@@ -466,3 +469,32 @@ func (s *SymDense) GrowSquare(n int) Matrix {
...
+func (s *SymDense) PowPSD(a Symmetric, pow float64) error {
+	dim := a.Symmetric()
+	s.reuseAs(dim)
+
+	var eigen EigenSym
+	ok := eigen.Factorize(a, true)
+	if !ok {
```
Comment (btracey) [inline mat64/symmetric.go]: Fixed.

File: mat64/symmetric_test.go (hunk 1 above)
Comment (btracey) [inline mat64/symmetric_test.go]: Done.

File: mat64/symmetric_test.go [hunk 4]
```diff
@@ -662,9 +662,32 @@ func TestPowPSD(t *testing.T) {
...
+		a := NewDense(dim, dim, nil)
+		for i := 0; i < dim; i++ {
+			for j := 0; j < dim; j++ {
+				a.Set(i, j, rnd.Float64())
+			}
+		}
+		var mat SymDense
+		mat.SymOuterK(1, a)
+
+		var sym SymDense
```
Comment (kortschak) [inline mat64/symmetric_test.go]:
```
const n = 5 // for some value of 5.
for pow := 2; pow < n; pow++ {
    ...
    sym.PowPSD(&mat, float64(pow))
    ...
    dense.Pow(&mat, pow)
    ...
}
```

**Pull request merged**